Frontend/library/src/Inputs/GamepadController.ts

```diff
@@ -1,59 +1,82 @@
 // Copyright Epic Games, Inc. All Rights Reserved.
 
-import { Logger } from '@epicgames-ps/lib-pixelstreamingcommon-ue5.5';
 import { StreamMessageController } from '../UeInstanceMessage/StreamMessageController';
-import { EventListenerTracker } from '../Util/EventListenerTracker';
 import { Controller } from './GamepadTypes';
 
 /**
- * The class that handles the functionality of gamepads and controllers
+ * Additional types for Window and Navigator
+ */
+declare global {
+    interface Window {
+        mozRequestAnimationFrame(callback: FrameRequestCallback): number;
+        webkitRequestAnimationFrame(callback: FrameRequestCallback): number;
+    }
+
+    interface Navigator {
+        webkitGetGamepads(): Gamepad[];
+    }
+}
+
+/**
+ * Gamepad layout codes enum
  */
+/* eslint-disable @typescript-eslint/no-duplicate-enum-values */
+export enum GamepadLayout {
+    RightClusterBottomButton = 0,
+    RightClusterRightButton = 1,
+    RightClusterLeftButton = 2,
+    RightClusterTopButton = 3,
+    LeftShoulder = 4,
+    RightShoulder = 5,
+    LeftTrigger = 6,
+    RightTrigger = 7,
+    SelectOrBack = 8,
+    StartOrForward = 9,
+    LeftAnalogPress = 10,
+    RightAnalogPress = 11,
+    LeftClusterTopButton = 12,
+    LeftClusterBottomButton = 13,
+    LeftClusterLeftButton = 14,
+    LeftClusterRightButton = 15,
+    CentreButton = 16,
+    // Axes
+    LeftStickHorizontal = 0,
+    LeftStickVertical = 1,
+    RightStickHorizontal = 2,
+    RightStickVertical = 3
+}
+/* eslint-enable @typescript-eslint/no-duplicate-enum-values */
+
 export class GamePadController {
     controllers: Array<Controller>;
-    requestAnimationFrame: (callback: FrameRequestCallback) => number;
-    toStreamerMessagesProvider: StreamMessageController;
+    streamMessageController: StreamMessageController;
 
-    // Utility for keeping track of event handlers and to unregister them.
-    private gamePadEventListenerTracker = new EventListenerTracker();
+    onGamepadConnectedListener: (event: GamepadEvent) => void;
+    onGamepadDisconnectedListener: (event: GamepadEvent) => void;
+    beforeUnloadListener: (event: Event) => void;
+    requestAnimationFrame: (callback: FrameRequestCallback) => number;
 
-    /**
-     * @param toStreamerMessagesProvider - Stream message instance
-     */
-    constructor(toStreamerMessagesProvider: StreamMessageController) {
-        this.toStreamerMessagesProvider = toStreamerMessagesProvider;
+    constructor(streamMessageController: StreamMessageController) {
+        this.streamMessageController = streamMessageController;
 
+        this.onGamepadConnectedListener = this.gamePadConnectHandler.bind(this);
```

Comment:
This class is a shocker for casing.

`gamePad`, `gamepad`, and `Gamepad` all seen here

Maybe `gamePad` ones can be fixed up. Those seem to be the worst abuse.